### PR TITLE
fixing share logic and other postreqtoblobber fixes

### PR DESCRIPTION
**Pull request by chauchausoup**

### CODE REVIEW

**Comment by chauchausoup** [comment]
Should be merged (if review approved) only after the share-feature-bug branch merge on explorer.

**Review by shaktals — APPROVED** [review]

**Comment by shaktals** [comment]
We need to remember to make a full round of tests, after we start using these changes. To make sure that nothing brakes (although it seems very unlikely).

File: index.js [hunk 1]
```diff
@@ -772,12 +772,12 @@ module.exports = {
         return list
     },
 
-    getFileMetaDataFromPath: async function (allocation_id, path, client_id) {
+    getFileMetaDataFromPath: async function (allocation_id, path, client_id,public_key) {
```
Comment (shaktals) [inline index.js:775]: I didn't notice this before, but it'd be nice to keep good formatting.
Same applies to all other lines:
```suggestion
    getFileMetaDataFromPath: async function (allocation_id, path, client_id, public_key) {
```

File: index.js [hunk 2]
```diff
@@ -1194,7 +1194,7 @@ async function submitTransaction(ae, toClientId, val, note, transaction_type) {
     var data = {};
     data.client_id = ae.id;
     data.transaction_value = val;
-    data.transaction_data = { note };
+    data.transaction_data = {note};
```
Comment (shaktals) [inline index.js:1197]: @chauchausoup to fix the `sendZCN` function I made this change on the previous lib version:

frim
`data.transaction_data = note;`
to
`data.transaction_data = { note };`

However, this breaks a great many other functions, as the backend endpoints treat this field differently.
I reverted the `explorer` to use the previous version of `js-client-sdk` to _unbreak_ stuff.

But, if we want to use this new version of the lib, we'd have to revert this change again to:

`data.transaction_data = note;`

**Pull request merged**